src/overlays/actors/ovl_Bg_Dblue_Elevator/z_bg_dblue_elevator.c

```diff
@@ -5,68 +5,373 @@
+void BgDblueElevator_Pause(BgDblueElevator* this, PlayState* play);
+void BgDblueElevator_SetupMove(BgDblueElevator* this);
+void BgDblueElevator_Move(BgDblueElevator* this, PlayState* play);
+
 ActorInit Bg_Dblue_Elevator_InitVars = {
-    /**/ ACTOR_BG_DBLUE_ELEVATOR,
-    /**/ ACTORCAT_BG,
-    /**/ FLAGS,
-    /**/ OBJECT_DBLUE_OBJECT,
-    /**/ sizeof(BgDblueElevator),
-    /**/ BgDblueElevator_Init,
-    /**/ BgDblueElevator_Destroy,
-    /**/ BgDblueElevator_Update,
-    /**/ BgDblueElevator_Draw,
+    ACTOR_BG_DBLUE_ELEVATOR,
+    ACTORCAT_BG,
+    FLAGS,
+    OBJECT_DBLUE_OBJECT,
+    sizeof(BgDblueElevator),
+    (ActorFunc)BgDblueElevator_Init,
+    (ActorFunc)BgDblueElevator_Destroy,
+    (ActorFunc)BgDblueElevator_Update,
+    (ActorFunc)BgDblueElevator_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B929EC[] = {
+static BgDblueElevatorStruct1 D_80B92960[4] = {
+    {
+        0,
+        ((s32(*)(BgDblueElevator*, PlayState*))BgDblueElevator_GetWaterFlowFromCeiling),
+        320.0f,
+        30,
+        1,
+        { 0, 0 },
+        1.0f,
+        0.1f,
+        6.0f,
+    },
+    {
+        1,
+        ((s32(*)(BgDblueElevator*, PlayState*))BgDblueElevator_GetWaterFlowFromCeiling),
+        195.0f,
+        30,
+        1,
+        { 0, 0 },
+        1.0f,
+        0.1f,
+        5.0f,
+    },
+    {
+        0,
+        ((s32(*)(BgDblueElevator*, PlayState*))BgDblueElevator_GetWaterFlowFromPipes),
+        280.0f,
+        30,
+        1,
+        { 0, 0 },
+        1.0f,
+        0.1f,
+        6.0f,
+    },
+    {
+        0,
+        ((s32(*)(BgDblueElevator*, PlayState*))BgDblueElevator_GetWaterFlowFromPipes),
+        280.0f,
+        30,
+        -1,
+        { 0, 0 },
+        1.0f,
+        0.1f,
+        6.0f,
+    },
+};
+
+static s16 D_80B929D0[4] = { -90, -90, 90, 90 };   // Related to splash x pos
+static s16 D_80B929D8[4] = { -100, 90, 90, -100 }; // Related to splash z pos
+static s8 sLargeRipplesLives[3] = { 0, 2, 4 };
+static s8 sSmallRipplesLives[6] = { 0, 1, 2, 3, 4, 5 };
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 250, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 250, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+void BgDblueElevator_CheckWaterBoxInfo(BgDblueElevator* this, PlayState* play2) {
+    PlayState* play = play2;
+    WaterBox* waterBox;
+    s32 bgId;
+
+    this->isWithinWaterBoxXZ =
+        WaterBox_GetSurfaceImpl(play, &play->colCtx, this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.z,
+                                &this->waterSurfaceYPos, &waterBox, &bgId);
+}
+
+void BgDblueElevator_SpawnRipplesAndSplashes(BgDblueElevator* this, PlayState* play) {
+    s32 i;
+    Vec3f ripplePos;
+    Vec3f splashPos;
+    s32 j;
+    f32 d1;
+    f32 d2;
+    f32 v1;
+    f32 v2;
+    f32 var_fs0;
+    s32 rn;
```

Comment:
```suggestion
    s32 rand;
```
Pretty common through the codebase to name this `rand`